File: 프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.java
class Solution {
    public String solution(String s) {
        StringBuilder answer = new StringBuilder();
        String[] parts = s.split(" ", -1);

        for (int i = 0; i < parts.length; i++) {
            StringBuilder sb = new StringBuilder();
            for (int j = 0; j < parts[i].length(); j++) {
                char ch = parts[i].charAt(j);
                if (j % 2 == 0) {
                    sb.append(Character.toUpperCase(ch));
                } else {
                    sb.append(Character.toLowerCase(ch));
                }
            }
            answer.append(sb);
            if (i < parts.length - 1) answer.append(" ");
        }
        return answer.toString();
    }
}

File: 프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.py
def solution(s):
    answer = ''
    c = 0
    for i in range(len(s)):
        if s[i] == ' ':
            answer+=s[i]
            c = -1
        elif c % 2 == 0:
            answer += s[i].upper()
        else:
            answer += s[i].lower()
        
        c+=1
    return answer